internal/keystore/vault/client.go

```diff
@@ -210,6 +213,9 @@ func (c *client) RenewToken(ctx context.Context, authenticate authFunc, secret *
 					if err == nil {
 						break
 					}
+					if resp, ok := err.(*vaultapi.ResponseError); ok && http.StatusBadRequest <= resp.StatusCode && resp.StatusCode < http.StatusInternalServerError {
```

Comment:
I generally prefer to use `http.StatusXXX`, but if you want to check >=400 && <500 then using numbers is more readable.